pypy/module/faulthandler/cintf.py

```diff
@@ -26,7 +26,7 @@ def direct_llexternal(*args, **kwargs):
 
 
 DUMP_CALLBACK = lltype.Ptr(lltype.FuncType(
-                     [rffi.INT, rffi.SIGNEDP, lltype.Signed], lltype.Void))
+                     [rffi.INT_real, rffi.INTPTR_TP, rffi.INTPTR_T], lltype.Void))
```

Comment:
The callback signature uses `intptr_t *`, but that still is not working here.

Reply:
Yeah, I see:
```
	pypy_module_faulthandler.c: In function 'pypy_g_Handler_enable':
	pypy_module_faulthandler.c:117:45: error: passing argument 1 of 'pypy_faulthandler_setup' from incompatible pointer type [-Wincompatible-pointer-types]
	  117 |         l_v215550 = pypy_faulthandler_setup(pypy_g__dump_callback);
	      |                                             ^~~~~~~~~~~~~~~~~~~~~
	      |                                             |
	      |                                             void (*)(int,  Signed *, Signed) {aka void (*)(int,  long int *, long int)}
	In file included from singleheader.h:141:
	/home/stefanor/pypy/pypy/module/faulthandler/faulthandler.h:11:65: note: expected 'pypy_faulthandler_cb_t' {aka 'void (*)(int,  int *, int)'} but argument is of type 'void (*)(int,  Signed *, Signed)' {aka 'void (*)(int,  long int *, long int)'}
	   11 | RPY_EXTERN char *pypy_faulthandler_setup(pypy_faulthandler_cb_t dump_callback);
	      |                                          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
	pypy_module_faulthandler.c: In function 'pypy_g_Handler_dump_traceback':
```